python/mpt/tests/src/mpt/test_trie_diff.py

```diff
@@ -0,0 +1,129 @@
+from collections import defaultdict
+from pathlib import Path
+from typing import Optional
+
+import pytest
+from ethereum.cancun.trie import LeafNode
+from ethereum.crypto.hash import keccak256
+from ethereum_types.bytes import Bytes, Bytes32
+from hypothesis import HealthCheck, given, settings
+from hypothesis import strategies as st
+
+from mpt.ethereum_tries import EthereumTrieTransitionDB
+from mpt.trie_diff import StateDiff
+from mpt.utils import AccountNode, decode_node
+
+
+@pytest.fixture
+def ethereum_trie_transition_db(data_path):
+    return EthereumTrieTransitionDB.from_json(data_path)
+
+
+@pytest.fixture(scope="session")
+def node_store(zkpi):
+    nodes = defaultdict(
+        lambda: None,
+        {
+            keccak256(Bytes.fromhex(node[2:])): decode_node(Bytes.fromhex(node[2:]))
+            for node in zkpi["witness"]["state"]
+        },
+    )
+    return nodes
+
+
+class TestTrieDiff:
+    @pytest.mark.parametrize(
+        "data_path", [Path("test_data/22081873.json")], scope="session"
+    )
+    def test_trie_diff(self, data_path, ethereum_trie_transition_db):
+        # Python
+        state_diff = StateDiff.from_json(data_path)
+        trie_diff = StateDiff.from_tries(ethereum_trie_transition_db)
+        assert trie_diff._main_trie == state_diff._main_trie
+        assert trie_diff._storage_tries == state_diff._storage_tries
+
+    @pytest.mark.parametrize(
+        "data_path", [Path("test_data/22081873.json")], scope="session"
+    )
+    @given(data=st.data())
+    @settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
```

Comment:
i don't think you need the healthcheck anymore if all the used fixtures are scoped session

Reply:
suggest remove pls